app/js/enhancedDate.js

```diff
@@ -1,5 +1,52 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function() {    
+    var myDate = new Date();    
+    myDate.setHours(0, 0, 0, 0);
+
+    return {    
+        getDate: function() {   
+            return myDate.toDateString();
+        },
+        setDate: function(newDate) {
+            if (isNaN(Date.parse(newDate))) {
```

Comment:
setDate can take either a Date object or a millisecond value. If either of those types of values are passed in, you want to pass them to the Date constructor and assign the result to myDate.